src/mmap.rs

```diff
@@ -448,10 +448,18 @@ impl GuestMemory for GuestMemoryMmap {
     }
 
     fn find_region(&self, addr: GuestAddress) -> Option<&GuestRegionMmap> {
-        for region in self.regions.iter() {
-            if addr >= region.start_addr() && addr <= region.end_addr() {
-                return Some(region);
+        let bs_region = self.regions.binary_search_by_key(&addr, |x| x.start_addr());
+        match bs_region {
+            Ok(x) => {
+                return Some(&self.regions[x]);
             }
+            Err(x) if x > 0 => {
```

Comment:
We might be able to replace the following block with:

```rust
if self.regions[x - 1].end_addr() >= addr {
    return Some(self.regions[x - 1]);
}
```

Off-topic: this also kinda reminds me that `end_addr()` might not be most suggestive name for a method that returns the address of the last byte from the region (instead of that of the following byte). Do you think the name is actually ok, and in line with usual conventions (maybe `last_addr()` is more appropriate)?

Reply:
No, I don't&mdash;`last_addr` is definitely better.